.circleci/config.yml

```diff
@@ -1,163 +1,61 @@
-      - store_test_results:
-          path: ~/coverage-reports
-      - store_artifacts:
-          path: ~/workdir/target/scala-2.12/coverage-report
-      - store_artifacts:
-          path: ~/workdir/target/scala-2.12/scoverage-report
-      - store_artifacts:
-          path: ~/workdir/target/scala-2.12/scoverage-data
-
-  publish:
-    <<: *default_java_job
-    steps:
-      - *restore_source_code
-      - *setup_dependencies_key
-      - *restore_dependencies
-      - *restore_build
-      - run:
-          name: Install AWS CLI
-          command: |
-            sudo apt -y update
-            sudo apt -y install python3-pip
-            sudo python3 -m pip install awscli
-      - run:
-          name: Setup AWS Credentials
-          command: ./scripts/setup-aws-credentials.sh
-      - run:
-          name: Retrieve GPG Keys
-          command: |
-            mkdir -p ~/.gnupg
-            aws s3 sync --profile shared-services --include '*.gpg' s3://$AWS_CREDENTIALS_BUCKET/gnupg/ ~/.gnupg
-      - run:
-          name: Publish to Sonatype
-          command: ./scripts/publish.sh 3.0.$CIRCLE_BUILD_NUM
+orbs:
+  codacy: codacy/base@0.3.12
 
 workflows:
   version: 2
-  test-and-publish:
+  compile_test_deploy:
     jobs:
-      - checkout_code
-      - sbt_dependencies:
+      - codacy/checkout_and_version:
+          write_sbt_version: true
+      - codacy/sbt:
+          name: populate_cache
+          cmd: sbt "set scalafmtUseIvy in ThisBuild := false;update"
+          persist_to_workspace: true
           requires:
-            - checkout_code
-      - compile:
+            - codacy/checkout_and_version
+      - codacy/sbt:
+          name: check_fmt
+          cmd: sbt ";scalafmt::test;test:scalafmt::test;sbt:scalafmt::test"
           requires:
-            - checkout_code
-            - sbt_dependencies
-      - test:
+            - populate_cache
+      - codacy/sbt:
+          name: compile
+          cmd: |
+            sbt +compile
+            sbt +test:compile
+          requires:
+            - check_fmt
+      - codacy/sbt:
+          name: test
+          cmd: |
+            sbt ";coverage;+test;coverageReport;coverageAggregate"
+
+            if [ -z "$CODACY_PROJECT_TOKEN" ]; then
+                echo "CODACY_PROJECT_TOKEN not found. Skipping send coverage to Codacy."
+            else
+                sbt codacyCoverage
+            fi
           requires:
             - compile
-      - publish:
+      - codacy/sbt:
+          name: publish
+          cmd: |
+            sudo apt -y update
+            sudo apt -y install python3-pip
+            sudo python3 -m pip install awscli
+            ./scripts/setup-aws-credentials.sh
+            mkdir -p ~/.gnupg
+            aws s3 sync --profile shared-services --include '*.gpg' s3://$AWS_CREDENTIALS_BUCKET/gnupg/ ~/.gnupg
+            ./scripts/publish.sh "$(cat .version)"
```

Comment:
You could use that orb: https://circleci.com/orbs/registry/orb/circleci/aws-cli

Reply:
I searched about that orb, and this `publish` step also uses `sbt`. So I guess it's preferred to install awscli there instead of adding the sbt installation. What do you think?

We also have AWS things on sbt orb. https://github.com/codacy/codacy-orbs/blob/master/src/jobs/sbt.yml . I guess we get the gpg keys using an sbt step, so this is kinda strange to me.